src/test/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSinkUnitTest.java

```diff
@@ -68,17 +83,48 @@
     @Test(dataProvider = "loadReadsBAM", groups = "spark")
     public void readsSinkTest(String inputBam, String outputFileName, String outputFileExtension) throws IOException {
         final File outputFile = createTempFile(outputFileName, outputFileExtension);
+        assertSingleShardedWritingWorks(inputBam, outputFile.getAbsolutePath());
+    }
+
+    @Test(dataProvider = "loadReadsBAM", groups = "spark")
+    public void readsSinkHDFSTest(String inputBam, String outputFileName, String outputFileExtension) throws IOException {
+        final String outputHDFSPath = MiniClusterUtils.getTempPath(cluster, outputFileName, outputFileExtension).toString();
+        assertSingleShardedWritingWorks(inputBam, outputHDFSPath);
+    }
+
+    @Test(dataProvider = "loadReadsBAM", groups = "spark")
+    public void testWritingToAnExistingFileHDFS(String inputBam, String outputFileName, String outputFileExtension) throws IOException {
```

Comment:
Ah yes, I missed that. Could you add in an assert that the file exists before the `assertSingleShardedWritingWorks()` call anyway?